### PR TITLE
NO-JIRA Avoid possible double instantiation of properties

### PR DESCRIPTION
**Pull request by michaelandrepearce**

### CODE REVIEW

**Comment by michaelandrepearce** [comment]
@jbertram nudge.

**Comment by jbertram** [comment]
I just created ARTEMIS-2099.  I'll update the commit message to use this when I merge it.  I'm just waiting for the PR build to finish.

File: artemis-core-client/src/main/java/org/apache/activemq/artemis/core/message/impl/CoreMessage.java [hunk 1]
```diff
@@ -551,12 +551,16 @@ public CoreMessage setUserID(UUID userID) {
     */
    protected TypedProperties checkProperties() {
       if (properties == null) {
-         TypedProperties properties = new TypedProperties();
-         if (buffer != null && propertiesLocation >= 0) {
-            final ByteBuf byteBuf = buffer.duplicate().readerIndex(propertiesLocation);
-            properties.decode(byteBuf, coreMessageObjectPools == null ? null : coreMessageObjectPools.getPropertiesDecoderPools());
+         synchronized (this) {
```
Comment (jbertram) [inline artemis-core-client/src/main/java/org/apache/activemq/artemis/core/message/impl/CoreMessage.java]: Shouldn't the synchronized go *inside* the null check?

Comment (michaelandrepearce) [reply artemis-core-client/src/main/java/org/apache/activemq/artemis/core/message/impl/CoreMessage.java]: This is double checked, thus once instantiated further calls don't incur the synchronized cost. it imperative the field is volatile for this, but i checked and properties is already volatile.

if (properties == null) {
         synchronized (this) {
            if (properties == null)

**Pull request closed without merging**